src/mcts/search.cc

```diff
@@ -541,22 +555,35 @@ void SearchWorker::GatherMinibatch() {
     // There was a collision. If limit has been reached, return, otherwise
     // just start search of another node.
     if (picked_node.is_collision) {
-      if (++collisions_found > search_->kAllowedNodeCollisions) return;
+      if (++collisions_found_ > search_->kAllowedNodeCollisions) return;
       continue;
     }
     ++nodes_found;
 
     // If node is already known as terminal (win/loss/draw according to rules
     // of the game), it means that we already visited this node before.
-    if (node->IsTerminal()) continue;
-
-    // Node was never visited, extend it.
-    ExtendNode(node);
-
-    // Only send non-terminal nodes to neural network
     if (!node->IsTerminal()) {
-      nodes_to_process_.back().nn_queried = true;
-      AddNodeToComputation(node);
+      // Node was never visited, extend it.
+      ExtendNode(node);
+
+      // Only send non-terminal nodes to neural network
+      if (!node->IsTerminal()) {
+        nodes_to_process_.back().nn_queried = true;
+        nodes_to_process_.back().is_cache_hit =
+            AddNodeToComputation(node, true);
+      }
+    }
+
+    // If out of order eval is enabled and the node to compute we added last
+    // doesn't require NN eval (i.e. it's a cache hit or terminal node), do
+    // out of order eval for it.
+    if (search_->kOutOfOrderEval) {
+      const bool is_terminal = node->IsTerminal();
+      if (is_terminal ||
+          (!is_terminal && nodes_to_process_.back().is_cache_hit)) {
```

Comment:
Isn't the `!is_terminal` here redundant with the first part of the condition?

IOW, couldn't this just be `if (is_terminal || nodes_to_process_.back().is_cache_hit) {`?

Reply:
Indeed, fixed.